cylc/flow/task_state.py

```diff
@@ -154,6 +155,77 @@
 }
 
 
+class RunMode:
+    """The possible run modes of a task/workflow."""
+
+    LIVE = 'live'
+    """Task will run normally."""
+
+    SIMULATION = 'simulation'
+    """Task will run in simulation mode."""
+
+    DUMMY = 'dummy'
+    """Task will run in dummy mode."""
+
+    SKIP = 'skip'
+    """Task will run in skip mode."""
+
+    WORKFLOW = 'workflow'
+    """Default to workflow run mode"""
+
+    MODES = {LIVE, SIMULATION, DUMMY, SKIP, WORKFLOW}
+
+    WORKFLOW_MODES = sorted(MODES - {WORKFLOW})
```

Comment:
I think we would want live mode to be first
```suggestion
    WORKFLOW_MODES = [LIVE, DUMMY, SIMULATION, SKIP]
```